Cargo.toml

```diff
@@ -12,20 +10,22 @@ repository = "https://github.com/quartiq/miniconf"
 keywords = ["settings", "serde", "no_std", "json", "mqtt"]
 categories = ["no-std", "config", "rust-patterns", "parsing"]
 rust-version = "1.65.0"
+resolver = "2"
 
 [dependencies]
 miniconf_derive = { path = "miniconf_derive" , version = "0.6" }
-serde-json-core = { git = "https://github.com/rust-embedded-community/serde-json-core.git" }
+serde-json-core = { git = "https://github.com/rust-embedded-community/serde-json-core.git" , optional = true}
 serde = { version = "1.0.120", features = ["derive"], default-features = false }
-log = "0.4"
-heapless = { version = "0.7", features = ["serde"] }
+log = {version = "0.4", optional = true}
+heapless = { version = "0.7", features = ["serde"], optional=true }
```

Comment:
We can remove the public re-exports on `heapless` and `serde_json_core` now I think

Reply:
The exports are already feature-dependent.

Reply:
Yeah, what I'm saying is that they should be removed entirely now. They're unnecessary and should be deprecated/removed from what I understand.

Reply:
Right. I'm not entirely sure about the serde_json_core errors though as they are still part of the API. OTOH they are accessible.